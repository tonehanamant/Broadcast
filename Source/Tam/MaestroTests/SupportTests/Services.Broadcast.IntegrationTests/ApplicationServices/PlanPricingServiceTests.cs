using ApprovalTests;
using ApprovalTests.Reporters;
using IntegrationTests.Common;
using NUnit.Framework;
using Services.Broadcast.ApplicationServices;
using Services.Broadcast.ApplicationServices.Plan;
using Tam.Maestro.Common.DataLayer;

namespace Services.Broadcast.IntegrationTests.ApplicationServices
{
    [TestFixture]
    public class PlanPricingServiceTests
    {
        private readonly IPlanPricingService _PlanPricingService = IntegrationTestApplicationServiceFactory.GetApplicationService<IPlanPricingService>();
        private readonly IPlanService _PlanService = IntegrationTestApplicationServiceFactory.GetApplicationService<IPlanService>();

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void RunNoDataTest()
        {
            using (new TransactionScopeWrapper())
            {
                var result = _PlanPricingService.Run(1196);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void RunTest()
        {
            using (new TransactionScopeWrapper())
            {
                var result = _PlanPricingService.Run(1198);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void RunWithHiatusDayTest()
        {
            using (new TransactionScopeWrapper())
            {
                var result = _PlanPricingService.Run(1199);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void RunSpotLengthTest()
        {
            using (new TransactionScopeWrapper())
            {
                var result = _PlanPricingService.Run(1197);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
            }
        }
    }
}
