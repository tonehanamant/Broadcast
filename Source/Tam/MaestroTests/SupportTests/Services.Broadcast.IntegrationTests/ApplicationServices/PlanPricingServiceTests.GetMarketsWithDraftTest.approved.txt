{
  "MarketDetails": [
    {
      "Budget": 5375.0,
      "Cpm": 16.888641014261082679716742100,
      "Impressions": 318.26125,
      "ImpressionsPercentage": 99.10011249,
      "MarketCode": 0,
      "MarketCoveragePercent": 10.0,
      "MarketName": "New York",
      "Rank": 1,
      "ShareOfVoiceGoalPercentage": 34.5,
      "Spots": 15,
      "Stations": 1
    },
    {
      "Budget": 185.0,
      "Cpm": 64.013840830449826989619377200,
      "Impressions": 2.89,
      "ImpressionsPercentage": 0.89988751,
      "MarketCode": 0,
      "MarketCoveragePercent": 0.328,
      "MarketName": "Portland-Auburn",
      "Rank": 79,
      "ShareOfVoiceGoalPercentage": 22.2,
      "Spots": 15,
      "Stations": 1
    }
  ],
  "PlanVersionId": 0,
  "PostingType": 2,
  "PricingJobId": null,
  "Totals": {
    "Budget": 5560.0,
    "CoveragePercent": 10.328,
    "Cpm": 17.3127148033831,
    "Impressions": 321.15125,
    "Markets": 2,
    "Spots": 30,
    "Stations": 2
  }
}